go.mod

```diff
@@ -25,6 +25,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware/v2 v2.0.0-rc.3
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.14.0
+	github.com/hashicorp/golang-lru/v2 v2.0.1
```

Comment:
Yeah, I considered using goburrow/cache, but it caches the result regardless of the result, so we would need to error when an object is not found to prevent the result from being cached. I didn't think that made sense so I preferred to use a cache directly like this.